### PR TITLE
Run test in cloud

### PR DESCRIPTION
**Pull request by going-confetti**

### CODE REVIEW

File: src-tauri/src/main.rs
Comment (going-confetti) [inline src-tauri/src/main.rs:1]: @oleiade there's some copy-paste/copilot output here, but most importantly, I wanted to get your feedback on storing the token/projectId:
1. Can (cloud) projectId be a part of the project metadata? 
2. What's the best way to store the access token?

Comment (oleiade) [reply src-tauri/src/main.rs:1]: Coincidentally I hit the same needs in a different context today. I was thinking of:
* creating an ApplicationConfig structure that the application state would hold. That would be a good place to store application wide configuration, such as a cloud token.
* creating a ProjectConfig structure that's would hold project specific configuration, and which we could serialize to JSON in the underlying filesystem place to persist over reboots.
* Exposing a get_configuration and set_configuration commands, as well as supporting passing a ProjectConfiguration when creating or updating a project.

I won't have time to do that tonight, but if you think that would fit your needs, I'd happily prioritize it Monday morning (or review your implementation if you have the time and/or will to take a stab at it in the meantime)

File: src/routes/test/edit/BlocksEditor.svelte [hunk 1]
```diff
@@ -239,15 +237,35 @@
 			});
 	}
 
+	function runTestInCloud(projectId: string) {
```
Comment (allansson) [inline src/routes/test/edit/BlocksEditor.svelte:240]: If you can merge this this evening, then I can add it into the new block editor.

File: src-tauri/tauri.conf.json
Comment (allansson) [inline src-tauri/tauri.conf.json:1]: Oooff, I just fixed merge conflicts in this file. Prettier is wreaking havoc.

File: src-tauri/tauri.conf.json [hunk 2]
```diff
@@ -1,69 +1,72 @@
...
+	},
+	"tauri": {
+		"allowlist": {
+			"all": false,
+			"dialog": {
+				"message": true
+			},
+			"shell": {
+				"open": true
+			}
```
Comment (allansson) [inline src-tauri/tauri.conf.json:21]: This is the only real change?

File: src-tauri/Cargo.toml [hunk 3]
```diff
@@ -17,7 +17,7 @@ tauri-build = { version = "1.5.0", features = [] }
 [dependencies]
 serde_json = "1.0"
 serde = { version = "1.0", features = ["derive"] }
-tauri = { version = "1.5.3", features = ["dialog-message"] }
+tauri = { version = "1.5.3", features = [ "shell-open", "dialog-message"] }
```
Comment (Llandy3d) [inline src-tauri/Cargo.toml:20]: I wouldn't go this route, I think we can opt to open a window and load that page similar to what we do with the local run ? what do you think 🤔

File: src-tauri/src/main.rs
Comment (Llandy3d) [reply src-tauri/src/main.rs:1]: For the access token I was thinking of delegating it to `k6 login cloud -t` that would save it under (for example on my macos machine): `/Users/llandy/Library/Application Support/loadimpact/k6/config.json`

But we can handle it all ourselves if we prefer and the token at the application level , while project_id at the project level makes sense. The model from theo looks perfect for this, I wonder what you had in mind while considering projectConfig, what else do you plan to store 🤔

**Review by allansson — COMMENTED** [review]
I can't get this to work. I see this in the Rust log:

```
ERRO[0001] (400/E5) Authentication error - could not query matching resource. Is the token and/or project_id correct? 
```

I've tried both a Grafana Stack token and my personal token. I've made sure the project id is correct

File: src-tauri/Cargo.toml (hunk 3 above)
Comment (going-confetti) [reply src-tauri/Cargo.toml:20]: To me, opening Grafana in the browser makes more sense. Here are some of the reasons I can think of right now:
* You probably have your Grafana credentials saved in the browser, so if you have to re-login, it's easier to do in the browser
* It makes sharing the link easier
* If the user opens the chart in Explore or decided to export results as PDF, a new tab is opened where they are directed to a different place in Grafana/app.k6.io - will this work and look okay-ish in a desktop window?

File: src-tauri/tauri.conf.json
Comment (Llandy3d) [reply src-tauri/tauri.conf.json:1]: let's try to get everything on main and do one final pass where we clean it up 🤔

File: src-tauri/tauri.conf.json (hunk 2 above)
Comment (going-confetti) [reply src-tauri/tauri.conf.json:21]: Yes

File: src-tauri/Cargo.toml (hunk 3 above)
Comment (Llandy3d) [reply src-tauri/Cargo.toml:20]: I think there are multiple ways to tackle those things (after all we do have the credentials for tests!) but that overall falls into the cloud integration category that is not a priority for this week beside running the test. 
So I'm convinced this is the best approach for now 👍

File: src-tauri/src/main.rs [hunk 4]
```diff
@@ -148,3 +157,40 @@ impl ApplicationState {
...
+#[tauri::command]
+async fn run_script_in_cloud(script: String, project_id: String) -> Result<String, String>{
+    let mut child = Command::new("k6")
+    .arg("cloud")
+    .arg("-")
+    .stdin(Stdio::piped())
+    .stdout(Stdio::piped())
+    .env("K6_CLOUD_TOKEN", std::env::var(String::from("K6_CLOUD_TOKEN")).unwrap_or(String::from("")))
+    .env("K6_CLOUD_PROJECT_ID", project_id)
+    .env("K6_CLOUD_NAME", "kroco6 script.js")
```
Comment (going-confetti) [inline src-tauri/src/main.rs:181]: There's [this issue](https://github.com/grafana/k6/issues/3077), which was a surprise as I could stream results to cloud without the test name, but couldn't _run_ test in cloud without it

**Review by allansson — APPROVED** [review]
Works! Great job!

**Pull request merged**